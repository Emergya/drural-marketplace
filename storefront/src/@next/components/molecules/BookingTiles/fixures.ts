export const resourceCalendarAvailability = [
  {
    date: "2021-12-27",
    available: true,
  },
  {
    date: "2021-12-28",
    available: false,
  },
  {
    date: "2021-12-29",
    available: true,
  },
  {
    date: "2021-12-30",
    available: false,
  },
  {
    date: "2021-12-31",
    available: true,
  },
  {
    date: "2022-01-01",
    available: false,
  },
  {
    date: "2022-01-02",
    available: false,
  },
  {
    date: "2022-01-03",
    available: true,
  },
  {
    date: "2022-01-04",
    available: false,
  },
  {
    date: "2022-01-05",
    available: true,
  },
  {
    date: "2022-01-06",
    available: false,
  },
  {
    date: "2022-01-07",
    available: true,
  },
  {
    date: "2022-01-08",
    available: false,
  },
  {
    date: "2022-01-09",
    available: false,
  },
  {
    date: "2022-01-10",
    available: false,
  },
  {
    date: "2022-01-11",
    available: false,
  },
  {
    date: "2022-01-12",
    available: false,
  },
  {
    date: "2022-01-13",
    available: false,
  },
  {
    date: "2022-01-14",
    available: false,
  },
  {
    date: "2022-01-15",
    available: false,
  },
  {
    date: "2022-01-16",
    available: false,
  },
  {
    date: "2022-01-17",
    available: true,
  },
  {
    date: "2022-01-18",
    available: false,
  },
  {
    date: "2022-01-19",
    available: true,
  },
  {
    date: "2022-01-20",
    available: false,
  },
  {
    date: "2022-01-21",
    available: true,
  },
  {
    date: "2022-01-22",
    available: false,
  },
  {
    date: "2022-01-23",
    available: false,
  },
  {
    date: "2022-01-24",
    available: true,
  },
  {
    date: "2022-01-25",
    available: false,
  },
  {
    date: "2022-01-26",
    available: true,
  },
  {
    date: "2022-01-27",
    available: false,
  },
  {
    date: "2022-01-28",
    available: true,
  },
  {
    date: "2022-01-29",
    available: false,
  },
  {
    date: "2022-01-30",
    available: false,
  },
  {
    date: "2022-01-31",
    available: true,
  },
  {
    date: "2022-02-01",
    available: false,
  },
  {
    date: "2022-02-02",
    available: true,
  },
  {
    date: "2022-02-03",
    available: false,
  },
  {
    date: "2022-02-04",
    available: true,
  },
];

export const resourceAvailabilityByDate = [
  {
    startTime: "01:45:00",
    endTime: "02:15:00",
  },
  {
    startTime: "02:15:00",
    endTime: "02:45:00",
  },
  {
    startTime: "02:45:00",
    endTime: "03:15:00",
  },
  {
    startTime: "07:45:00",
    endTime: "08:15:00",
  },
  {
    startTime: "08:15:00",
    endTime: "08:45:00",
  },
];
