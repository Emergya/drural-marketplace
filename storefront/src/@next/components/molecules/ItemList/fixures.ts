export const collections = [
  {
    __typename: "Collection",
    id: "1",
    name: "Collection 1",
    slug: "collection-1",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "2",
    name: "Collection 2",
    slug: "collection-2",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "3",
    name: "Collection 3",
    slug: "collection-3",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "4",
    name: "Collection 4",
    slug: "collection-4",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "5",
    name: "Collection 5",
    slug: "collection-5",
    backgroundImage: null,
  },
  {
    __typename: "Collection",
    id: "6",
    name: "Collection 6",
    slug: "collection-6",
    backgroundImage: null,
  },
];
