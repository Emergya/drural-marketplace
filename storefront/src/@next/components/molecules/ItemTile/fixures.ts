export const collection = {
  __typename: "Collection",
  id: "1",
  name: "Collection 1",
  slug: "collection-1",
  backgroundImage: null,
};

export const category = {
  __typename: "Category",
  id: "1a",
  name: "Category 1",
  slug: "category-1",
  iconId: null,
};
