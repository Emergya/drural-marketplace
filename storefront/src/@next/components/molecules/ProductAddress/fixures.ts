import { ProductAddressCountry } from "@drural/sdk";
import { ProductDetails_address } from "@drural/sdk/lib/fragments/gqlTypes/ProductDetails";

export const PRODUCT_ADDRESS: ProductDetails_address = {
  __typename: "ProductAddressType",
  street: "Pascal",
  postalCode: "28034",
  locality: "Madrid",
  region: "Madrid",
  country: ProductAddressCountry.ES,
  latitude: 37.37301,
  longitude: -5.74951,
};
