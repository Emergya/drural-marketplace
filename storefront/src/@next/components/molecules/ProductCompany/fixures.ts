import { ProductDetails_company } from "@drural/sdk/lib/fragments/gqlTypes/ProductDetails";

export const PRODUCT_COMPANY: ProductDetails_company = {
  __typename: "CompanyType",
  id: "Q29tcGFueVR5cGU6MQ==",
  imageUrl: "/media/companies/dRural_-_Marketp_11.png",
  name: "Comany 01",
  publicName: "Comany 01",
  rating: 2,
  products: {
    __typename: "ProductCountableConnection",
    totalCount: 3,
  },
};
