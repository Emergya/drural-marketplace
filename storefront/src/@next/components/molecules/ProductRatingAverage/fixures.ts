import { IProps } from "./types";

export const DEFAULT_PROPS: IProps = {
  rating: 3,
  reviewsCount: 8,
};
