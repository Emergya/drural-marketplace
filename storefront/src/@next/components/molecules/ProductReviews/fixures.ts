import { GetServiceReviews_product } from "@pages/ReviewsPage/gqlTypes/GetServiceReviews";

export const PRODUCT_REVIEWS: Partial<GetServiceReviews_product> = {
  consumedByUser: true,
  rating: 2,
  reviews: {
    totalCount: 3,
    pageInfo: {
      __typename: "PageInfo",
      hasNextPage: false,
      endCursor: "123123123123",
    },
    __typename: "ProductRatingCountableConnection",
    edges: [
      {
        __typename: "ProductRatingCountableEdge",
        node: {
          __typename: "ProductRating",
          id: "123",
          createdByUser: true,
          user: "Pepito Garcia",
          createdAt: "2021-11-25T07:34:22Z",
          rating: 4,
          comment:
            "This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. ",
        },
      },
      {
        __typename: "ProductRatingCountableEdge",
        node: {
          __typename: "ProductRating",
          id: "123",
          createdByUser: true,
          user: "Pepito Garcia",
          createdAt: "2021-11-25T07:34:22Z",
          rating: 4,
          comment:
            "This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. ",
        },
      },
      {
        __typename: "ProductRatingCountableEdge",
        node: {
          __typename: "ProductRating",
          id: "123",
          createdByUser: true,
          user: "Pepito Garcia",
          createdAt: "2021-11-25T07:34:22Z",
          rating: 4,
          comment:
            "This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. ",
        },
      },
    ],
  },
};
