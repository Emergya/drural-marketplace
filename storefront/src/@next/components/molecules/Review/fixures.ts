import { GetServiceReviews_product_reviews_edges_node } from "@pages/ReviewsPage/gqlTypes/GetServiceReviews";

export const PRODUCT_REVIEW: GetServiceReviews_product_reviews_edges_node = {
  __typename: "ProductRating",
  id: "123",
  createdByUser: true,
  user: "Pepito Garcia",
  createdAt: "2021-11-25T07:34:22Z",
  rating: 4,
  comment:
    "This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. This service is good. ",
};
