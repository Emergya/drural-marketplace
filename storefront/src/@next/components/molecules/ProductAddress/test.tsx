import "jest-styled-components";

import { mount, ReactWrapper } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import { dRuralTheme } from "@styles";

import { ProductAddress } from ".";
import { PRODUCT_ADDRESS } from "./fixures";

describe("<ProductInfo />", () => {
  // 1. Variables
  let wrapper: ReactWrapper;

  // 2. Wrapper
  beforeEach(() => {
    wrapper = mount(<ProductAddress address={PRODUCT_ADDRESS} />);
  });

  // 3. Tests
  it("exists", () => {
    expect(wrapper.exists()).toEqual(true);
  });

  it("contains provided data", () => {
    expect(wrapper.text()).toContain(PRODUCT_ADDRESS.postalCode);
    expect(wrapper.text()).toContain(PRODUCT_ADDRESS.street);
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={dRuralTheme}>
          <ProductAddress address={PRODUCT_ADDRESS} />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
