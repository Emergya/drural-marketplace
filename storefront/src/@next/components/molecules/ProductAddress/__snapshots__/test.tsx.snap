// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductInfo /> renders correctly 1`] = `
.c0 {
  padding-bottom: 26px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 1rem;
}

.c3 svg {
  width: 100%;
}

.c4 {
  background-color: #C1BEC8;
  height: 192px;
  width: 100%;
}

.c2 {
  font-size: 0.8125rem;
  margin-left: 0.5rem;
}

@media (max-width:540px) {
  .c0 {
    padding-bottom: 24px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <svg
      fill="currentColor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18,4.48a8.45,8.45,0,0,0-12,12l5.27,5.28a1,1,0,0,0,1.42,0L18,16.43A8.45,8.45,0,0,0,18,4.48ZM16.57,15,12,19.59,7.43,15a6.46,6.46,0,1,1,9.14,0ZM9,7.41a4.32,4.32,0,0,0,0,6.1,4.31,4.31,0,0,0,7.36-3,4.24,4.24,0,0,0-1.26-3.05A4.3,4.3,0,0,0,9,7.41Zm4.69,4.68a2.33,2.33,0,1,1,.67-1.63A2.33,2.33,0,0,1,13.64,12.09Z"
      />
    </svg>
    <p
      className="c2"
    >
      28034, Pascal
    </p>
  </div>
  <div
    className="c3"
  >
    <div
      className="c4"
    />
  </div>
</div>
`;
